api_tests/tests/dry/rfc003_schema.ts

```diff
@@ -0,0 +1,220 @@
+    const swapEntity = swapResponse.body as Entity;
+
+    expect(swapEntity, message).to.be.jsonSchema(sirenJsonSchema);
+    expect(swapEntity.properties, message).to.be.jsonSchema(
+        swapPropertiesJsonSchema
+    );
+}
+
+describe("Dry - rfc003 schema", () => {
+    it("get-all-swaps-is-valid-siren", async function() {
+        await twoActorTest(async function({ alice }) {
+            const res = await request(alice.cndHttpApiUrl()).get("/swaps");
+
+            expect(res.body).to.be.jsonSchema(sirenJsonSchema);
+        });
+    });
+    it("get-single-swap-is-valid-siren", async function() {
+        await twoActorTest(async function({ alice, bob }) {
+            // Alice send swap request to Bob
+            await alice.cnd.postSwap(await createDefaultSwapRequest(bob));
+
+            const aliceSwapEntity = await alice
+                .pollCndUntil("/swaps", body => body.entities.length > 0)
+                .then(
+                    body => body.entities[0] as EmbeddedRepresentationSubEntity
+                );
+
+            await assertValidSirenDocument(
+                aliceSwapEntity,
+                alice,
+                "[Alice] Response for GET /swaps/rfc003/{} is a valid siren document and properties match the json schema"
+            );
+
+            const bobsSwapEntity = await bob
+                .pollCndUntil("/swaps", body => body.entities.length > 0)
+                .then(
+                    body => body.entities[0] as EmbeddedRepresentationSubEntity
+                );
+            await assertValidSirenDocument(
+                bobsSwapEntity,
+                bob,
+                "[Bob] Response for GET /swaps/rfc003/{} is a valid siren document and properties match the json schema"
+            );
+        });
+    });
+
+    it("get-single-swap-contains-link-to-rfc", async function() {
+        await twoActorTest(async function({ alice, bob }) {
+            // Alice send swap request to Bob
+            await alice.cnd.postSwap(await createDefaultSwapRequest(bob));
+
+            const aliceSwapEntity = await alice
+                .pollCndUntil("/swaps", body => body.entities.length > 0)
+                .then(
+                    body => body.entities[0] as EmbeddedRepresentationSubEntity
+                );
+
+            const protocolLink = aliceSwapEntity.links.find((link: Link) =>
+                link.rel.includes("describedBy")
+            );
+
+            expect(protocolLink).to.be.deep.equal({
+                rel: ["describedBy"],
+                class: ["protocol-spec"],
+                type: "text/html",
+                href:
+                    "https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.adoc",
+            });
+        });
+    });
+});
+
+// ******************************************** //
+// RFC003 Swap Reject                           //
+// ******************************************** //
```

Comment:
I find this a bit odd but I guess you had your reason to add it :)

Reply:
I find it more readable? Happy to remove it if people don't like it :)

Reply:
I don't personally like these sort of comments, they do not add anything that the function names already add and are prone to becoming stale.  I am always suspicious of codebases that use these style of comments, I actually tried to get these sort of comments removed from rustc (amusingly the issue got a thumbs up, a thumbs down, and a confused :) https://github.com/rust-lang/rust/issues/62601